drivers/flash/soc_flash_numaker.c

```diff
@@ -0,0 +1,290 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ *  Copyright (c) 2023 Nuvoton Technology Corporation.
+ */
+
+#define DT_DRV_COMPAT nuvoton_numaker_fmc
+
+#include <string.h>
+#include <errno.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/flash.h>
+#include <zephyr/logging/log.h>
+#include "flash_priv.h"
+#include <NuMicro.h>
+
+LOG_MODULE_REGISTER(flash_numaker, CONFIG_FLASH_LOG_LEVEL);
+
+#define SOC_NV_FLASH_NODE DT_INST(0, soc_nv_flash)
+
+extern int32_t FMC_EraseBank(uint32_t u32_bank_addr);
+
+struct flash_numaker_data {
+	FMC_T *fmc;
+	struct k_sem write_lock;
+	uint32_t flash_block_base;
+};
+
+static const struct flash_parameters flash_numaker_parameters = {
+#if DT_NODE_HAS_PROP(SOC_NV_FLASH_NODE, write_block_size)
+	.write_block_size = DT_PROP(SOC_NV_FLASH_NODE, write_block_size),
```

Comment:
Could you use `DT_PROP_OR` instead of the conditional?

Reply:
Yes, DT_PROP_OR is better.